Automattic/woocommerce-payments
# HACK week: Improve express checkout layout for better experience

#### Changes proposed in this Pull Request
This PR aims to improve the ECE loading experience on the shortcode checkout page, by removing the layout shifts. We moved to flex layout for better control, as well as control the appearance through opacity instead of displaying and hiding the wrapping container.

#### Testing instructions

> 💡 There's plenty of comparison operation of Before & After. You can achieve this by switching between `develop` and `improve/express-checkout-container-loading` (the way it's described below) but I found it easier to just spin up two separate JN sites with `develop` and `improve/express-checkout-container-loading` packages. 

> We need to check both old ECE and tokenized-ECE. Feel free to start with one and once the steps are completed - switch to another and perform them again

##### Only Apple Pay/Google Pay enabled
0. Check out `develop`
1. Navigate to the shortcode checkout
2. Notice about three layout shifts when ECE is being loaded
3. Save the page (we'll need it for comparison later)
4. Check out `improve/express-checkout-container-loading` 
5. Open the shortcode checkout page in a separate tab and confirm the ECE is loaded smoothly
6. Compare both tabs in terms of ECE container placement and confirm that there are no changes between them (except from this)
7. Navigate to the cart and confirm the ECE container is loaded as expected 


##### Both Apple Pay/Google Pay & WooPay are enabled
0. Check out `develop`
1. Navigate to the shortcode checkout
2. Notice the pre-rendered purple WooPay button appears on both `develop` 
3. Check out `improve/express-checkout-container-loading`
4. Confirm the more smooth ECE loading (checkout + classic cart)
5. Confirm `develop` and `improve/express-checkout-container-loading`  are equal (except for this change) on checkout & classic cart
    * the most important things are top margin of 4px for WooPay and 12px for ECE, but comparing before & after should be enough to catch potential difference

##### Only WooPay enabled
* perform similar steps for WooPay only, it should look similar as on `develop` (no extra space between WooPay button and the OR separator
* check both cart and checkout of the shortcode type



## WooPay + ECE Before

https://github.com/user-attachments/assets/6b0447fb-2bd2-491e-a99e-21b2c71f33b4

## WooPay + ECE After

https://github.com/user-attachments/assets/e6752371-72ea-4246-8442-7e6359a88f02


## ECE Before

https://github.com/user-attachments/assets/2c04823c-44a9-43db-b329-f6f63146a488



## ECE After


https://github.com/user-attachments/assets/3834a50b-0e64-4576-8d6b-ae05b82b1db9


Stretched ECE & WooPay buttons (aligned with how WooPay button only was displayed up until today)
Before
<img width="1000" alt="image" src="https://github.com/user-attachments/assets/be1180cc-26a6-4447-a1f6-e96a65a3e2c7" />

After
<img width="1004" alt="image" src="https://github.com/user-attachments/assets/c049ff77-c300-451c-972f-1f5c37e9b76a" />




**Post merge**
